internal/xcuit/config.go

```diff
@@ -0,0 +1,136 @@
+	"gopkg.in/yaml.v2"
+)
+
+var supportedDeviceTypes = []string{"ANY", "PHONE", "TABLET"}
+
+// Project represents the xcuit project configuration.
+type Project struct {
+	config.TypeDef `yaml:",inline"`
+	ConfigFilePath string             `yaml:"-" json:"-"`
+	Sauce          config.SauceConfig `yaml:"sauce,omitempty" json:"sauce"`
+	Xcuit          Xcuit              `yaml:"xcuit,omitempty" json:"xcuit"`
+	Suites         []Suite            `yaml:"suites,omitempty" json:"suites"`
+	Artifacts      config.Artifacts   `yaml:"artifacts,omitempty" json:"artifacts"`
+}
+
+// Xcuit represents xcuit apps configuration.
+type Xcuit struct {
+	App     string `yaml:"app,omitempty" json:"app"`
+	TestApp string `yaml:"testApp,omitempty" json:"testApp"`
+}
+
+// TestOption represents the xcuit test filter options configuration.
+type TestOption struct {
+	Class []string `yaml:"class,omitempty" json:"class"`
+}
+
+// Suite represents the xcuit test suite configuration.
+type Suite struct {
+	Name        string     `yaml:"name,omitempty" json:"name"`
+	Devices     []Device   `yaml:"devices,omitempty" json:"devices"`
+	TestOptions TestOption `yaml:"testOptions,omitempty" json:"testOptions"`
+}
+
+// Device represents device configuration.
+type Device struct {
+	ID              string  `yaml:"id,omitempty" json:"id"`
+	Name            string  `yaml:"name,omitempty" json:"name"`
+	PlatformVersion string  `yaml:"platformVersion,omitempty" json:"platformVersion"`
+	PlatformName    string  `yaml:"platformName,omitempty" json:"platformName"`
+	Options         Options `yaml:"options,omitempty" json:"options"`
+}
+
+// Options represents device options configuration.
+type Options struct {
+	CarrierConnectivity bool   `yaml:"carrierConnectivity,omitempty" json:"carrierConnectivity"`
+	DeviceType          string `yaml:"deviceType,omitempty" json:"deviceType"`
+	Private             bool   `yaml:"private,omitempty" json:"private"`
+}
+
+// FromFile creates a new xcuit Project based on the filepath cfgPath.
+func FromFile(cfgPath string) (Project, error) {
+	var p Project
+
+	f, err := os.Open(cfgPath)
+	if err != nil {
+		return Project{}, fmt.Errorf("failed to locate project config: %v", err)
+	}
+	defer f.Close()
+
+	if err := yaml.NewDecoder(f).Decode(&p); err != nil {
+		return Project{}, fmt.Errorf("failed to parse project config: %v", err)
+	}
+	p.ConfigFilePath = cfgPath
+
+	if p.Sauce.Concurrency < 1 {
+		// Default concurrency is 2
+		p.Sauce.Concurrency = 2
+	}
+
+	return p, nil
+}
+
+// Validate validates basic configuration of the project and returns an error if any of the settings contain illegal
+// values. This is not an exhaustive operation and further validation should be performed both in the client and/or
+// server side depending on the workflow that is executed.
+func Validate(p Project) error {
+	if p.Xcuit.App == "" {
+		return errors.New("missing path to app .ipa")
+	}
+	if !strings.HasSuffix(p.Xcuit.App, ".ipa") {
```

Comment:
Shouldn't we support `.app` artifacts ?

Reply:
I would use `.ipa` for now and create a separate ticket for supporting `.app`.

Reply:
@alexplischke @vrunoa what are your thoughts about adding `.app` as a separate feature?

Reply:
I think that'd be a nice to have! But I also don't think it's necessary for this MVP, as we already have feature parity with the old runner from what I can tell 💡 So, something to tackle as a separate story.